src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupDetails/EnvironmentGroupDetails.tsx

```diff
@@ -0,0 +1,440 @@
+  const { showSuccess, showError, clear } = useToaster()
+
+  const formikRef = useRef<FormikProps<EnvironmentGroupResponseDTO>>()
+
+  const [selectedTabId, setSelectedTabId] = useState<EnvironmentGroupDetailsTab>(
+    EnvironmentGroupDetailsTab[EnvironmentGroupDetailsTab[defaultTo(sectionId, 'CONFIGURATION')]]
+  )
+  const [selectedEnvs, setSelectedEnvs] = useState<EnvironmentResponse[]>([])
+  const [yamlHandler, setYamlHandler] = useState<YamlBuilderHandlerBinding | undefined>()
+  const [selectedView, setSelectedView] = useState<SelectedView>(SelectedView.VISUAL)
+  const [updateLoading, setUpdateLoading] = useState(false)
+  const [firstLoad, setFirstLoad] = useState(true)
+
+  const { data, loading, error, refetch } = useGetEnvironmentGroup({
+    queryParams: {
+      accountIdentifier: accountId,
+      orgIdentifier,
+      projectIdentifier
+    },
+    envGroupIdentifier: defaultTo(environmentGroupIdentifier, '')
+  })
+
+  useEffect(() => {
+    if (!loading && firstLoad) {
+      setFirstLoad(false)
+    }
+  }, [loading])
+
+  const handleTabChange = (tabId: EnvironmentGroupDetailsTab) => {
+    updateQueryParams({
+      sectionId: EnvironmentGroupDetailsTab[EnvironmentGroupDetailsTab[tabId]]
+    })
+    setSelectedTabId(tabId)
+  }
+
+  const { data: environmentGroupSchema } = useGetYamlSchema({
+    queryParams: {
+      entityType: 'EnvironmentGroup',
+      projectIdentifier,
+      orgIdentifier,
+      accountIdentifier: accountId,
+      scope: getScopeFromDTO({ accountIdentifier: accountId, orgIdentifier, projectIdentifier })
+    }
+  })
+
+  const handleModeSwitch = React.useCallback(
+    (view: SelectedView) => {
+      if (view === SelectedView.VISUAL) {
+        const yaml = defaultTo(yamlHandler?.getLatestYaml(), '')
+        const yamlVisual = parse(yaml).environmentGroup as EnvironmentGroupResponseDTO
+        if (yamlVisual) {
+          formikRef.current?.setValues({
+            ...omit(cleanData(yamlVisual) as EnvironmentGroupResponseDTO)
+          })
+        }
+      }
+      setSelectedView(view)
+    },
+    [yamlHandler?.getLatestYaml, data]
+  )
+
+  const onUpdate = async (values: EnvironmentGroupResponseDTO) => {
+    setUpdateLoading(true)
+    try {
+      const body = yamlStringify(
+        JSON.parse(
+          JSON.stringify({
+            environmentGroup: { ...values }
+          })
+        )
+      )
+
+      const response = await updateEnvironmentGroupPromise({
+        body,
+        queryParams: {
+          accountIdentifier: accountId,
+          orgIdentifier,
+          projectIdentifier
+        },
+        envGroupIdentifier: environmentGroupIdentifier,
+        requestOptions: { headers: { 'Content-Type': 'application/yaml' } }
+      })
+
+      if (response.status === 'SUCCESS') {
+        clear()
+        showSuccess(getString('common.environmentGroup.updated'))
+        refetch()
+      } else {
+        throw response
+      }
+    } catch (e: any) {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232281)
```suggestion
    } catch (e: unknown) {
```